### DISCUSSION TITLE
On memory usage grow during sort spill

### DISCUSSION THREAD

**Discussion by yjshen** [apache/datafusion#9528] [General]
Through examining the current implementation of multi-column sort's spill-to-disk strategies, I find we are asking for more memory during spill, which I think is worth discussing:

During the spill, `Rows` are created for comparison efficiency for each in-memory RecordBatch. Considering why we spill in the first place, does this `Rows` optimization increase the possibility of execution failure due to memory shortage?

Please share your insights or corrections if I misunderstand any part of the code.

**Comment by Omega359** [comment]
The comments on [this issue](https://github.com/apache/arrow-datafusion/issues/9417) may possibly be relevant (not 100% sure though)
